GitHubProjects.ps1

```diff
@@ -0,0 +1,751 @@
+        the background, enabling the command prompt to provide status information.
+        If not supplied here, the DefaultNoStatus configuration property value will be used.
+
+    .EXAMPLE
+        Set-GitHubProject -OwnerName Microsoft -RepositoryName PowerShellForGitHub -Name TestProject -State Closed
+
+        Set the 'TestProject' project state to closed for the Microsoft\PowerShellForGitHub
+        repository.
+
+    .EXAMPLE
+        Set-GitHubProject -OrganizationName Microsoft -Name TestProject -Description 'Updated description'
+
+        Updates the description of the 'TestProject' that's owned by the Microsoft organization.
+
+    .EXAMPLE
+        Set-GitHubProject -Uri https://github.com/Microsoft/PowerShellForGitHub -Name TestProject -State Closed
+
+        Updates the state to closed for the Microsoft\PowerShellForGitHub repository project
+        by using the uri.
+
+    .EXAMPLE
+        Set-GitHubProject -UserName GitHubUser -Name TestProject' -State Closed
+
+        Updates the state to closed for the user project.
+
+#>
+    [CmdletBinding(
+        SupportsShouldProcess,
+        DefaultParameterSetName = 'Elements'
+        )]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSShouldProcess", "", Justification = "Methods called within here make use of PSShouldProcess, and the switch is passed on to them inherently.")]
+    param(
+        [Parameter(Mandatory, ParameterSetName = 'Elements')]
+        [string] $OwnerName,
+
+        [Parameter(Mandatory, ParameterSetName = 'Elements')]
+        [string] $RepositoryName,
+
+        [Parameter(Mandatory, ParameterSetName = 'Uri')]
+        [string] $Uri,
+
+        [Parameter(Mandatory, ParameterSetName = 'Organization')]
+        [string] $OrganizationName,
+
+        [Parameter(Mandatory, ParameterSetName = 'User')]
+        [string] $UserName,
+
+        [Parameter(Mandatory, ParameterSetName = 'Project')]
+        [int64] $Project,
+
+        [Parameter()]
+        [string] $Name,
+
+        [Parameter()]
+        [string] $Description,
+
+        [Parameter()]
+        [ValidateSet('Open', 'Closed')]
+        [string] $State,
+
+        [Parameter()]
+        [ValidateSet('Read', 'Write', 'Admin', 'None')]
+        [string] $OrganizationPermission,
+
+        [Parameter()]
+        [bool] $Private,
+
+        [Parameter()]
+        [string] $AccessToken,
+
+        [Parameter()]
+        [switch] $NoStatus
+    )
+    Write-InvocationLog
+
+    $telemetryProperties = @{}
+
+    $uriFragment = [String]::Empty
+    $apiDescription = [String]::Empty
+
+    if ($PSBoundParameters.ContainsKey('Project'))
+    {
+        $projectObj = Get-GitHubProject -Project $Project
+    }
+    else
+    {
+        $projectDetails = @{}
+        foreach ($k in $($PSBoundParameters.Keys))
+        {
+            if($k -in ('OwnerName','RepositoryName','OrganizationName','UserName','Name','Uri')) {
+                $projectDetails[$k] = $PSBoundParameters[$k]
```

Comment:
So in this case, for the `Set-` function the only option would be to pass in the ProjectId. I understand your point on keeping it as close to API functionality, just making sure before I remove that code :) 

Currently I'm using `Get-GitHubProject` to get the project so I have more information about it.  Using that information to decide whether we can set the `private` or `organization_permission`.

Would the preferred approach in this case to just pass the options selected through to the API call, and then if it's not valid the error would come from `Invoke-GHRestMethod` rather than me handling it?

Reply:
Went ahead and made these changes.